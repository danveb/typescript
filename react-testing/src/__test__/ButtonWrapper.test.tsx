import { render, screen, fireEvent } from "@testing-library/react";
import { ButtonWrapper } from "../ButtonWrapper";

describe("test ButtonWrapper component", () => {
  // smoke test
  test("renders without crashing", () => {
    render(<ButtonWrapper title="Add Item" />); 
  });

  // snapshot test
  test("creates a snapshot", () => {
    const { asFragment } = render(
      <ButtonWrapper title="Add Item" />
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  test("handles onClick", () => {
    const onClick = jest.fn(); 
    render(<ButtonWrapper onClick={onClick} title="Add Item" />); 
    const buttonElement = screen.getByText("Add Item"); 
    fireEvent.click(buttonElement); 
    expect(buttonElement).toBeInTheDocument();  
    expect(onClick).toHaveBeenCalledTimes(1); 
  }); 
});