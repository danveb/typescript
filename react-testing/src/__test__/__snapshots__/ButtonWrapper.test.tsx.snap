// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ButtonWrapper component creates a snapshot 1`] = `
<DocumentFragment>
  <button>
    Add Item
  </button>
</DocumentFragment>
`;
